lib/ansible/modules/cloud/amazon/aws_secret_facts.py

```diff
@@ -0,0 +1,203 @@
+    returned: always
+    type: dict
+automatically_after_days:
+    description: Specifies the number of days between automatic scheduled rotations of the secret.
+    returned: always
+    type: int
+last_rotate_date:
+    description: The last date and time that the rotation process for this secret was invoked.
+    returned: always
+    type: string
+last_changed_date:
+    description: The last date and time that this secret was modified in any way.
+    returned: always
+    type: string
+last_access_date:
+    description: The last date that this secret was accessed. This value is truncated to midnight of the date and therefore shows only the date, not the time.
+    returned: always
+    type: string
+deleted_date:
+    description:
+      - The date and time on which this secret was deleted. Not present on active secrets. The secret can be recovered until the number of days
+        in the recovery window has passed, as specified in the RecoveryWindowInDays parameter of the DeleteSecret operation.
+    returned: always
+    type: string
+tags:
+    description: The list of user-defined tags that are associated with the secret.
+    returned: always
+    type: list
+secret_versions_to_stages:
+    description: A list of all of the currently assigned SecretVersionStage staging labels and the SecretVersionId that each is attached to.
+    returned: always
+    type: dict
+'''
+
+import os
+import re
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict
+
+try:
+    from botocore.exceptions import BotoCoreError, ClientError
+except ImportError:
+    pass  # handled by AnsibleAWSModule
+
+
+def camel_list_to_snake_list(list):
+    conv_list = []
+    for i in list:
+        conv_list.append(camel_dict_to_snake_dict(i))
+    return conv_list
+
+
+def main():
+    module = AnsibleAWSModule(
+        argument_spec={
+            'name': dict(type='str'),
+            'prefix': dict(type='str'),
+            'suffix': dict(type='str'),
+            'regex': dict(type='str'),
+        },
+        supports_check_mode=True,
+    )
+
+    result = {
+        'changed': True,
+        'desired_secrets': []
+    }
+
+    client = module.client('secretsmanager')
+
+    kwargs = {}
+    all_secrets = []
+
+    try:
+        while True:
+            response = client.list_secrets(**kwargs)
+            for i in response['SecretList']:
+                all_secrets.append(i)
+            try:
+                kwargs['nextToken'] = response['nextToken']
+            except KeyError:
+                break
+    except (BotoCoreError, ClientError) as e:
+        module.fail_json_aws(e, msg="Failed to look up secret data")
+
+    if module.params.get('name'):
+        for i in all_secrets:
+            if i['Name'] == module.params.get('name'):
+                result['desired_secrets'].append(camel_dict_to_snake_dict(i))
+    elif module.params.get('prefix'):
```

Comment:
The `elif` system here has a couple problems:

- If a user provides prefix, suffix, and regex options, we can have a scenario where their regex option is ignored because they provided name, which is earlier in the if/elif block. 
- If a secret matches several params (name and prefix, for example) it could be included twice if the `if/elif` is changed to a series of `if` statements. So some deduplication is needed.

Reply:
Or, mutually_exclusive could be used and allowing multiple options could be a later feature request. It's too bad list_secrets doesn't take a Filters argument.

Reply:
I decided to roll with `mutually_exclusive` since that was more in the spirit of what I was going for here.  I just wasn't aware that was available at the time I wrote this.